### PR TITLE
Fixes #3540 Omit unwanted links / buttons (v.2)

### LINKED ISSUE #3540: Omit unwanted links / buttons
There are few places where the unwanted links still appear, I believe those should be removed / modified.

- [x] "Subscribe to this thread" - appears for logged in users, but not for guests. Fine. But "Subscribe to this forum" appears for all; however clicking on it as a guest will show 'No permission' page.
- [x] "Ban this user in MCP / ACP" appears in every member's profile even if they are already banned. That can be removed or changed for already banned members to "Modify Ban / Unban".
- [x] "Open Buddy List" appear for those who don't have buddies.
- ~~"Mark All Forums Read" appears even if there is not a single unread post.~~ (deferred. lightbulbs buggy).
- [x] "Show Forum Team" link appears in Memberlist even if the Forum Team listing page is disabled.
- [x] Find all posts / Find all threads in profile even when there is no post / thread made by user. ~~Same for reputation details link. Same for Reputation report of Reputation-less user.~~ (Skipped reputation link as reputation can be 0 in case of equal positive and negative, thus; reputation 0 can also have report).
- [ ] Empty forum shows thread filter and all good thread controls along with postbit frame.
- ~~Showthread-> Moderator Tools : "Open/Close Thread" and "Stick/Unstick Thread" should be specific as per state like "Approve/Unapprove".~~ ( covered and to be addressed by #3654 )
- [x] PM / email links in own post / profile.
- [x] Report buttons for unreportable posts.

### PR DESCRIPTION
**Pull request by effone**
Second phase attempt to fix #3540
_(finger crossed)_

### CODE REVIEW

File: global.php [hunk 1]
```diff
@@ -529,14 +529,19 @@
 	// Format the welcome back message
 	$lang->welcome_back = $lang->sprintf($lang->welcome_back, build_profile_link(htmlspecialchars_uni($mybb->user['username']), $mybb->user['uid']), $lastvisit);
 
-	$searchlink = '';
+	$buddylink = $searchlink = $pmslink = '';
+
+    if(!empty($mybb->user['buddylist']))
```
Comment (euantorano) [inline global.php]: Indentation here doesn't seem right.

File: member.php [hunk 2]
```diff
@@ -2335,8 +2335,14 @@
 	$findposts = $findthreads = '';
 	if($mybb->usergroup['cansearch'] == 1)
 	{
-		eval("\$findposts = \"".$templates->get("member_profile_findposts")."\";");
-		eval("\$findthreads = \"".$templates->get("member_profile_findthreads")."\";");
+		if((int)$memprofile['postnum'])
```
Comment (euantorano) [inline member.php]: This cast seems like it's redundant?

Comment (effone) [reply member.php]: Done

File: global.php (hunk 1 above)
Comment (effone) [reply global.php]: Done

**Pull request merged**